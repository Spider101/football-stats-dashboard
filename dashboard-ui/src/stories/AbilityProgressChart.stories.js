import AbilityProgressChart from '../components/AbilityProgressChart';
import { getPlayerAbilityData } from './utils/storyDataGenerators';

export default {
    component: AbilityProgressChart,
    title: 'Components/PlayerProgressionView/AbilityProgressChart',
    excludeStories: /.*Data$/,
    argTypes: {
        abilityData: {
            name: 'Ability Progression Data',
            control: { type: 'object' }
        }
    },
    parameters: {
        docs: {
            description: {
                component: 'UI Component for displaying a bar chart documenting a player\'s Ability progress over the'
                + ' last _N_ months.'
            }
        }
    }
};

const Template = args => <AbilityProgressChart { ...args } />;

export const MultipleDataPoints = Template.bind({});
MultipleDataPoints.args = {
    abilityData: getPlayerAbilityData()
};

export const SingleDataPoint = Template.bind({});
SingleDataPoint.args = {
    abilityData: getPlayerAbilityData(1)
};