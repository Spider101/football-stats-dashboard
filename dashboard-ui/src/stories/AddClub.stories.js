import AddClub from '../widgets/AddClub';

export default {
    component: AddClub,
    title: 'Widgets/ClubPageView/AddClub',
    argTypes: { addClubAction: { control: { disable: true } } },
    decorators: [
        Story => (
            <div style={{ display: 'flex', justifyContent: 'center' }}>
                <Story />
            </div>
        )
    ]
};

const Template = args => <AddClub {...args} />;

export const Success = Template.bind({});
Success.args = {
    addClubAction: _ => null
};

export const FormFailure = Template.bind({});
FormFailure.args = {
    addClubAction: _ => 'Failed to add new club!'
};