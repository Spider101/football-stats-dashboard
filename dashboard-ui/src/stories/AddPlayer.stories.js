import AddPlayer from '../widgets/AddPlayer';

export default {
    component: AddPlayer,
    title: 'Widgets/SquadHubView/AddPlayer',
    argTypes: { addPlayerAction: { control: { disable: true } } },
    decorators: [
        Story => (
            <div style={{ display: 'flex', justifyContent: 'center' }}>
                <Story />
            </div>
        )
    ]
};

const Template = args => <AddPlayer {...args} />;

export const Success = Template.bind({});
Success.args = {
    addPlayerAction: _ => null
};

export const FormFailure = Template.bind({});
FormFailure.args = {
    addPlayerAction: _ => 'Failed to add new player!'
};