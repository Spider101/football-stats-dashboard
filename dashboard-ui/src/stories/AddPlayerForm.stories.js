import { action } from '@storybook/addon-actions';
import { LookupDataContextProvider } from '../context/LookupDataProvider';
import { getLookupDataHandlers } from '../mocks/handlers';

import AddPlayerForm from '../widgets/AddPlayerForm';

export default {
    component: AddPlayerForm,
    title: 'Widgets/SquadHubView/AddPlayerForm',
    decorators: [
        Story => (
            <LookupDataContextProvider>
                <Story />
            </LookupDataContextProvider>
        )
    ]
};

const getFormMetadataAtStep = () => ({
    formData: { name: '', age: '', country: '' },
    formValidations: {},
    handleChangeFn: action('field was changed')
});

const Template = args => <AddPlayerForm {...args} />;
export const Default = Template.bind({});
Default.args = {
    getFormMetadataAtStep,
    stepIdx: 0
};

Default.parameters = {
    msw: {
        handlers: getLookupDataHandlers()
    }
};