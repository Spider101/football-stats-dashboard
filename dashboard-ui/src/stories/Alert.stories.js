import React from 'react';

import Alert from '../components/Alert';

export default {
    component: Alert,
    title: 'Components/Globals/Alert'
};

const Template = args => <Alert { ...args } />;

export const Success = Template.bind({});
Success.args = {
    severity: 'success',
    text: 'This is a success message!'
};

export const Error = Template.bind({});
Error.args = {
    severity: 'error',
    text: 'This is an error message!'
};