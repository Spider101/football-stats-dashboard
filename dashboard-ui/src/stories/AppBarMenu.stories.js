import React from 'react';
import { action } from '@storybook/addon-actions';

import AppBarMenu from '../components/AppBarMenu';
import { AuthContextProvider } from '../context/authProvider';
import { QueryClient, QueryClientProvider } from 'react-query';

export default {
    component: AppBarMenu,
    title: 'Components/Globals/AppBarMenu',
    parameters: {
        docs: {
            description: {
                component: 'UI Component at the top of the application for displaying information pertinent to the'
                + ' current screen. Currently, it hosts functionality like displaying the team\'s name, a search bar'
                + 'and a user profile badge UI element. It also contains a button for expanding and closing'
                + 'the side bar menu.'
            }
        }
    },
    decorators: [
        Story => (
            <QueryClientProvider client={new QueryClient()}>
                <AuthContextProvider>
                    <Story />
                </AuthContextProvider>
            </QueryClientProvider>
        )
    ]
};

const Template = args => <AppBarMenu {...args} />;

export const Default = Template.bind({});
Default.args = {
    menu: {
        title: 'Manchester United',
        teamColor: 'red'
    },
    onClickHandler: action('open-menu'),
    isOpen: false
};

export const MenuOpened = Template.bind({});
MenuOpened.args = {
    ...Default.args,
    isOpen: true
};
