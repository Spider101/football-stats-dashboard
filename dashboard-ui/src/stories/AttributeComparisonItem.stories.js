import React from 'react';
import AttributeComparisonItem from '../components/AttributeComparisonItem';
import faker from 'faker';

export const getAttrComparisonItemData = () => ({
    attrComparisonItem: {
        attrValues: [{
            name: faker.name.lastName(1),
            data: [ Math.round(Math.random() * 20) ]
        }, {
            name: faker.name.lastName(1),
            data: [ -1 * Math.round(Math.random() * 20) ]
        }],
        label: faker.hacker.noun()
    },
    isHighlighted: false
});

export default {
    component: AttributeComparisonItem,
    title: 'Components | PlayerComparisonView/AttributeComparisonTable/AttributeComparisonItem',
    excludeStories: /.*Data$/,
};

export const Default = () => <AttributeComparisonItem { ...getAttrComparisonItemData() } />;

export const Highlighted = () => <AttributeComparisonItem { ...getAttrComparisonItemData() } isHighlighted={ true }/>;
