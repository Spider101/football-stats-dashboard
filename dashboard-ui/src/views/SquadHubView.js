import { useState, useRef, useEffect, useMemo } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';

import flagCodes from '../flagCodes';
import { squadTableHeaderDisplayTypeMap, moraleIconsMap } from '../utils';
import SortableTable from '../widgets/SortableTable';
import TableFilterControl from '../components/TableFilterControl';

const useStyles = makeStyles(theme => ({
    fab: {
        position: 'fixed',
        bottom: theme.spacing(4),
        right: theme.spacing(4)
    }
}));

// TODO: define a more appropriate method for this
const getSortValueForForm = matchRatingsList => matchRatingsList[0];

const getFlagImageFromCountryName = countryName => `https://flagcdn.com/w40/${flagCodes[countryName]}.png`;

const buildHeaderDataForSquadTable = headerNames =>
    headerNames
        .filter(name => name !== 'playerId')
        .map(name => ({
            id: name,
            type: squadTableHeaderDisplayTypeMap[name]
        }));

const buildRowDataForSquadTable = players => {
    return players.map(player => {
        const keys = Object.keys(player);
        const playerId = player.playerId;
        const keysToFocusOn = keys.filter(key => key !== 'playerId');
        return Object.entries(_.pick(player, keysToFocusOn)).map(([key, value]) => {
            let row = null;
            switch (key) {
            case 'wages':
                row = { id: key, type: 'string', data: '$' + value + 'K' };
                break;
            case 'nationality':
                row = {
                    id: key,
                    type: 'image',
                    data: getFlagImageFromCountryName(value),
                    metadata: { sortValue: value }
                };
                break;
            case 'morale':
                row = {
                    id: key,
                    type: 'icon',
                    data: moraleIconsMap.find(entity => entity.morale === value).icon,
                    metadata: { sortValue: value }
                };
                break;
            case 'form':
                row = {
                    id: key,
                    type: 'chart',
                    data: {
                        type: 'bar',
                        series: [
                            {
                                name: 'Match Rating',
                                data: value
                            }
                        ]
                    },
                    metadata: { sortValue: getSortValueForForm(value) }
                };
                break;
            case 'name':
                row = {
                    id: key,
                    type: 'link',
                    data: value,
                    metadata: { playerId }
                };
                break;
            default:
                row = { id: key, type: isNaN(value) ? 'string' : 'number', data: value };
                break;
            }
            return row;
        });
    });
};

const filterColumns = (originalHeaders, filteredRowData, selectedColumns) => {
    const headers = originalHeaders.filter(header => selectedColumns.includes(header.id));
    const updatedRows = filteredRowData.map(rowData =>
        rowData.filter(cellData => selectedColumns.includes(cellData.id))
    );

    return {
        headers,
        // return empty list if all rows are just empty lists themselves
        rows: updatedRows.every(row => row.length === 0) ? [] : updatedRows
    };
};

const filterRowsByRole = (originalRowData, roles) =>
    originalRowData.filter(rowData => {
        const roleData = rowData.find(cell => cell.id === 'role');
        return roles.includes(roleData.data);
    });

export default function SquadHubView({ players, addPlayerWidget }) {
    const classes = useStyles();
    const [playerRoles, setPlayerRoles] = useState([]);

    const initialSquadHubTableHeaders = buildHeaderDataForSquadTable(
        Object.keys(players.length === 0 ? squadTableHeaderDisplayTypeMap : players[0])
    );
    const allSquadHubTableHeaderNames = initialSquadHubTableHeaders.map(header => header.id);
    const [columnNames, setColumnNames] = useState(allSquadHubTableHeaderNames);

    const allPlayerRoles = useRef([]);
    useEffect(() => {
        // get all the distinct player roles in the dataset
        allPlayerRoles.current = [...new Set(players.map(player => player.role))];

        setPlayerRoles(allPlayerRoles.current);
    }, [players]);

    const handleChange = changeHandler => event => changeHandler(event.target.value);

    // we just need to calculate this once when the component is mounted
    const rowData = useMemo(() => buildRowDataForSquadTable(players), [players]);

    const filteredRowData = filterRowsByRole(rowData, playerRoles);

    return (
        <>
            <Grid container spacing={2}>
                <Grid item xs={6}>
                    <TableFilterControl
                        currentValues={columnNames}
                        handleChangeFn={handleChange(setColumnNames)}
                        allPossibleValues={allSquadHubTableHeaderNames}
                        allValuesSelectedLabel='All Columns'
                        inputLabelText='Configure Columns'
                        labelIdFragment='configure-columns'
                    />
                </Grid>
                <Grid item xs={6}>
                    <TableFilterControl
                        currentValues={playerRoles}
                        handleChangeFn={handleChange(setPlayerRoles)}
                        allPossibleValues={allPlayerRoles.current}
                        allValuesSelectedLabel='All Player Roles'
                        inputLabelText='Filter Players By Role'
                        labelIdFragment='filter-rows'
                        customStyles={{ float: 'right' }}
                    />
                </Grid>
                <Grid item xs={12}>
                    <SortableTable {...filterColumns(initialSquadHubTableHeaders, filteredRowData, columnNames)} />
                </Grid>
            </Grid>
            <div className={classes.fab}>
                {addPlayerWidget}
            </div>
        </>
    );
}

SquadHubView.propTypes = {
    players: PropTypes.arrayOf(
        PropTypes.shape({
            playerId: PropTypes.number,
            name: PropTypes.string,
            nationality: PropTypes.string,
            role: PropTypes.string,
            wages: PropTypes.number,
            form: PropTypes.arrayOf(PropTypes.number),
            morale: PropTypes.string,
            currentAbility: PropTypes.number
        })
    ),
    addPlayerWidget: PropTypes.node
};