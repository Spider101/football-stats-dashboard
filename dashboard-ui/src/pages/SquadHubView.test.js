import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { Default, NoPlayers } from '../stories/SquadHubView.stories';

const getSanitizedProps = ({ players, ...rest }) => {
    return {
        ...rest,
        players: players.map(player => {
            // removing form because apex charts is not working with react-testing library for some reason
            // eslint-disable-next-line no-unused-vars
            const { form, ...rest } = player;
            return rest;
        })
    };
};

it('should render the player data when passed in', () => {
    render(
        <MemoryRouter>
            <Default {...getSanitizedProps(Default.args)} />
        </MemoryRouter>
    );

    const { players } = Default.args;
    const rows = screen.getAllByRole('row');
    // 1 extra row for the table header
    expect(rows.length).toEqual(players.length + 1);
});

it('should always render add player widget', () => {
    // taking snapshot here mainly to track any breaking changes due to add player widget
    // not passing any player data to avoid snapshot changing due to randomly generated data
    const { container } = render(
        <MemoryRouter>
            <NoPlayers {...NoPlayers.args} />
        </MemoryRouter>
    );

    expect(screen.queryByLabelText('add')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});