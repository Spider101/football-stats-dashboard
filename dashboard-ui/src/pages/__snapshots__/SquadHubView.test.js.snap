// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should always render add player widget 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        class="MuiFormControl-root makeStyles-formControl-30"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
          id="configure-columns-select-label"
        >
          Configure Columns
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="configure-columns-select-label configure-columns-checkbox"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            id="configure-columns-checkbox"
            role="button"
            tabindex="0"
          >
            <div
              class="makeStyles-chips-31"
            >
              <div
                class="MuiChip-root makeStyles-chip-32"
              >
                <span
                  class="MuiChip-label"
                >
                  All Columns
                </span>
              </div>
            </div>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value="name,nationality,role,wages,form,morale,current_ability"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        class="MuiFormControl-root makeStyles-formControl-30"
        style="float: right;"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          id="filter-rows-select-label"
        >
          Filter Players By Role
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="filter-rows-select-label filter-rows-checkbox"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            id="filter-rows-checkbox"
            role="button"
            tabindex="0"
          >
            <span>
              â€‹
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiTableContainer-root"
      >
        <table
          class="MuiTable-root makeStyles-table-33"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                aria-sort="ascending"
                class="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-37 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-38 MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root WithStyles(ForwardRef(TableSortLabel))-root-40 MuiTableSortLabel-active WithStyles(ForwardRef(TableSortLabel))-active-41"
                  role="button"
                  tabindex="0"
                >
                  Name
                  <span
                    class="makeStyles-visuallyHidden-36"
                  >
                    sorted ascending
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon WithStyles(ForwardRef(TableSortLabel))-icon-42 MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-37 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-38 MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root WithStyles(ForwardRef(TableSortLabel))-root-40"
                  role="button"
                  tabindex="0"
                >
                  Nationality
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon WithStyles(ForwardRef(TableSortLabel))-icon-42 MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-37 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-38 MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root WithStyles(ForwardRef(TableSortLabel))-root-40"
                  role="button"
                  tabindex="0"
                >
                  Role
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon WithStyles(ForwardRef(TableSortLabel))-icon-42 MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-37 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-38 MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root WithStyles(ForwardRef(TableSortLabel))-root-40"
                  role="button"
                  tabindex="0"
                >
                  Wages
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon WithStyles(ForwardRef(TableSortLabel))-icon-42 MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-37 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-38 MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root WithStyles(ForwardRef(TableSortLabel))-root-40"
                  role="button"
                  tabindex="0"
                >
                  Form
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon WithStyles(ForwardRef(TableSortLabel))-icon-42 MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-37 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-38 MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root WithStyles(ForwardRef(TableSortLabel))-root-40"
                  role="button"
                  tabindex="0"
                >
                  Morale
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon WithStyles(ForwardRef(TableSortLabel))-icon-42 MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root WithStyles(ForwardRef(TableCell))-root-37 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-38 MuiTableCell-alignRight"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root WithStyles(ForwardRef(TableSortLabel))-root-40"
                  role="button"
                  tabindex="0"
                >
                  Current Ability
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon WithStyles(ForwardRef(TableSortLabel))-icon-42 MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          />
        </table>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg"
        >
          <div
            class="MuiTypography-root MuiTypography-h5"
            style="text-align: center; padding: 5%;"
          >
            No Data is available for display. Please select one or more rows!
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-fab-29"
  >
    <button
      aria-label="add"
      class="MuiButtonBase-root MuiFab-root MuiFab-secondary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
